src/app/(with-header-sidebar)/mydashboard/hooks/useApi.ts

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect, useCallback } from 'react';
+import { AxiosResponse, AxiosError } from 'axios';
+import axiosInstance from '../utils/axiosInstance';
+
+type UseApiFetchReturnType<T> = {
+  data: T | null;
+  loading: boolean;
+  error: AxiosError | null;
+  refetch: () => Promise<void>;
+};
+
+type RequestOptions<
+  TParams = Record<string, unknown>,
+  TBody = Record<string, unknown>,
+> = {
+  method: 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+  params?: TParams;
+  body?: TBody;
+};
+
+export default function useApi<
+  T,
+  TParams = Record<string, unknown>,
+  TBody = Record<string, unknown>,
+>(
+  url: string,
+  options: RequestOptions<TParams, TBody>
+): UseApiFetchReturnType<T> {
+  const [data, setData] = useState<T | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<AxiosError | null>(null);
+
+  const fetchData = useCallback(async () => {
+    setLoading(true);
```

Comment:
```suggestion
if (loading) return;
setLoading(true);
```
이미 로딩 중인 경우 중복 호출 방지하기 위해 이런 처리가 있으면 좋을 것 같습니다.

Reply:
@najitwo 왜인지?모르겠는데 gpt 한테 리펙토링 해달라고하면 쟤를 없애더라구요...?
useAPi custom hook 찾아봐도 loading 중이면 return하는 코드는 없어서 먼가 이유가 있지 않을까 싶긴한데
요거 TODO 주석으로 해놓고 고려사항으로 넣어두겠습니다! 👍